pydantic_core/core_schema.py

```diff
@@ -181,24 +181,26 @@ def simple_ser_schema(type: ExpectedSerializationTypes) -> SimpleSerSchema:
     return SimpleSerSchema(type=type)
 
 
-class GeneralPlainSerializerFunction(Protocol):  # pragma: no cover
-    def __call__(self, __input_value: Any, __info: SerializationInfo) -> Any:
-        ...
-
-
-class FieldPlainSerializerFunction(Protocol):  # pragma: no cover
-    def __call__(self, __model: Any, __input_value: Any, __info: FieldSerializationInfo) -> Any:
-        ...
-
+# (__input_value: Any) -> Any
+GeneralPlainNoInfoSerializerFunction = Callable[[Any], Any]
+# (__input_value: Any, __info: FieldSerializationInfo) -> Any
+GeneralPlainInfoSerializerFunction = Callable[[Any, SerializationInfo], Any]
+# (__model: Any, __input_value: Any) -> Any
+FieldPlainNoInfoSerializerFunction = Callable[[Any, Any], Any]
+# (__model: Any, __input_value: Any, __info: FieldSerializationInfo) -> Any
+FieldPlainInfoSerializerFunction = Callable[[Any, Any, FieldSerializationInfo], Any]
+SerializerFunction = Union[
+    GeneralPlainNoInfoSerializerFunction,
+    GeneralPlainInfoSerializerFunction,
+    FieldPlainNoInfoSerializerFunction,
+    FieldPlainInfoSerializerFunction,
+]
 
-class GeneralPlainSerializerFunctionSchema(TypedDict):
-    type: Literal['general']
-    function: GeneralPlainSerializerFunction
 
-
-class FieldPlainSerializerFunctionSchema(TypedDict):
-    type: Literal['field']
-    function: FieldPlainSerializerFunction
+class SerializerFunctionSchema(TypedDict):
+    type: Literal['general', 'field']
+    info_arg: bool
```

Comment:
We should either make 4 types or 2 bools (`is_field` or something like that).

Reply:
happy to switch to two bools.